packages/agent-utils/src/agent/basicFunctionCallLoop.ts

```diff
@@ -23,21 +27,32 @@ export async function* basicFunctionCallLoop<TContext extends { llm: LlmApi, cha
 
     if (response.function_call) {
       const { name, arguments: args } = response.function_call;
-      const result = await executeAgentFunction(name, args, context, agentFunctions);
+      const { result, functionCalled } = await executeAgentFunction(name, args, context, agentFunctions);
```

Comment:
`name` already contains the name of the function

Reply:
I thought so too, but it is equal to `executeScript` which is not enough information. Instead we need the name + args (i.e. `executeScript({ namespace: "agent.onGoalAchieved" ... })`)

Reply:
ah, yea, good point